test/types/peg.test-d.ts

```diff
@@ -0,0 +1,330 @@
+import * as peggy from "../..";
+import tsd from "tsd";
+import {expectType} from "tsd";
+import formatter from "tsd/dist/lib/formatter";
+import {readFileSync} from "fs";
+import {join} from "path";
+
+
+const src = readFileSync(
+  join(__dirname, "..", "..", "examples", "fizzbuzz.peggy"),
+  "utf8"
+);
+
+describe("peg.d.ts", () => {
+  it("executes a grammar", () => {
+    expectType<string>(src);
+    expect(src.length).toBeGreaterThan(0);
+
+    const parser = peggy.generate(src);
+    expectType<peggy.Parser>(parser);
+
+    let res = parser.parse("1\n");
+    expectType<any>(res);
+    expect(res).toStrictEqual([1]);
+
+    res = parser.parse("buzz\n11\nfizz\n", { start: 10 });
+    expect(res).toStrictEqual([ "buzz", 11, "fizz" ]);
+  });
+
+  it("takes a valid tracer", () => {
+    const parser = peggy.generate(src, { trace: true });
+    expectType<peggy.Parser>(parser);
+
+    const res = parser.parse(" /**/ 1\n", {
+      startRule: "top",
+      tracer: {
+        trace(event) {
+          expectType<peggy.ParserTracerEvent>(event);
+          expectType<"rule.enter" | "rule.match" | "rule.fail">(event.type);
+          expectType<string>(event.rule);
+          switch (event.type) {
+            case "rule.enter":
+              expectType<peggy.LocationRange>(event.location);
+              break;
+            case "rule.match":
+              expectType<peggy.LocationRange>(event.location);
+              expectType<any>(event.result);
+              break;
+          }
+        }
+      }
+    })
+  });
+
+  it("takes an output and grammarSource", () => {
+    const p1 = peggy.generate(src, { output: "parser", grammarSource: "src" });
+    expectType<peggy.Parser>(p1);
+
+    const p2 = peggy.generate(src, { output: "source" });
```

Comment:
Why second test doesn't take a `grammarSource`?
```suggestion
    const p2 = peggy.generate(src, { output: "source", grammarSource: "src" });
  });
```

Reply:
Willfix, with something not a string.